rocAL/rocAL_pybind/amd/rocal/plugin/pytorch.py

```diff
@@ -193,15 +170,15 @@ def __next__(self):
 
         else:
             if(self.loader._oneHotEncoding == True):
-                self.loader.GetOneHotEncodedLabels(self.labels, self.device)
-                self.labels_tensor = self.labels.view(-1, self.bs, self.loader._numOfClasses).long()
+                print("Support for OneHotEncodedLabels is not added")
+                exit(0)
             else:
                 if self.display:
                     for i in range(self.bs):
                         img = (self.out)
                         draw_patches(img[i], i, 0)
-                self.loader.getImageLabels(self.labels)
-                self.labels_tensor = self.labels.long()
+                self.labels = self.loader.rocalGetImageLabels()
+                self.labels_tensor = self.labels_tensor.copy_(torch.from_numpy(self.labels)).long()
```

Comment:
Is deepcopy required here?
Can you check COCO pipeline if it is done there?

Reply:
This is required as we are creating the tensor on the device